lib/game_manager.rb

```diff
@@ -24,4 +24,96 @@ def games_of_season(season)
   def find_game_ids_for_season(season)
     games_of_season(season).map {|game| game.game_id }
   end
+
+
+#-------------GameStatistics
+
+  def highest_total_score
+    result = @games.max_by do |game|
+      game.away_goals.to_i + game.home_goals.to_i
+    end
+    result.away_goals.to_i + result.home_goals.to_i
+  end
+
+  def lowest_total_score
+    result = @games.min_by do |game|
+      game.away_goals.to_i + game.home_goals.to_i
+    end
+      result.away_goals.to_i + result.home_goals.to_i
+  end
+
+  def count_of_games_by_season
+    games_by_season_index = {}
+    games_by_season.each do |season, games|
+      games_by_season_index[season] = games.length
+    end
+    games_by_season_index
+  end
+
+  def games_by_season
+    result = {}
+    games.each do |game|
+      if result[game.season] == nil
+        result[game.season] = [game]
+      else
+        result[game.season] << game
+      end
+    end
+    result
+  end
+
+  def average_goals_per_game
+    (total_goals.to_f / total_number_of_games).round(2)
+  end
+
+  def total_goals
+    goal_count = 0
+    games.each do |game|
+      goal_count += game.home_goals.to_i
+      goal_count += game.away_goals.to_i
+    end
+    return goal_count
+  end
+
+  def total_number_of_games
+    game_count = 0
+    games.each do |game|
+      game_count += 1
+    end
+    return game_count
+  end
+
+  def average_goals_by_season
+    season_hash = get_total_goals
+    season_hash.each do |season, goals|
+      count_of_games_by_season.each do |season_games, games|
+        if season_games == season
+          average = goals.to_f / games.to_f
+          season_hash[season] = (average).round(2)
+        end
+      end
+    end
+  end
+
+  def get_total_goals
+    season_hash_with_goals = Hash.new
+    season_hash_with_goals = create_season_hash.each do |season, total_goals|
+      @games.each do |game|
+        if game.season == season
+          total_goals += game.away_goals.to_i
+          total_goals += game.home_goals.to_i
+        end
+      end
+      season_hash_with_goals[season] = total_goals
+    end
+  end
+
+  def create_season_hash
```

Comment:
This might be personal preference, but I'm not crazy about this method name. It doesn't really describe what the return value of this method is to me. I think something like games by season would be more descriptive of what it's doing!